pipelinewise/fastsync/commons/tap_postgres.py

```diff
@@ -348,18 +349,33 @@ def fetch_current_log_pos(self):
             raise Exception('Logical replication not supported before PostgreSQL 9.4')
 
         try:
-            # Create replication slot.
-            self.create_replication_slot()
+            # Create replication slot and obtain the oldest LSN
+            # that it can deliver.
+            oldest_lsn = self.create_replication_slot()
         except psycopg2.errors.DuplicateObject:
-            # Replication slot exists already so continue.
-            pass
+            # Replication slot exists already so just get the
+            # oldest LSN which the slot can deliver.
+            oldest_lsn = self.get_confirmed_flush_lsn()
 
         # Close replication slot dedicated connection
         self.primary_host_conn.close()
 
-        lsn = self.get_current_lsn()
+        current_lsn = self.get_current_lsn()
 
-        return {'lsn': lsn, 'version': 1}
+        if self.connection_config.get('replica_host'):
+            # Ensure that the newest LSN availiable on the replica is newer than
+            # the oldest LSN which is deliverable by the replication slot.
+            time_waited = 0  # seconds
+            wait_period = 1  # seconds
+            max_time_waited = 60  # seconds
+            while oldest_lsn > current_lsn:
+                time.sleep(wait_period)
+                time_waited += wait_period
```

Comment:
Strictly speaking this should probably use `time.time` to check the time passed, though I'm guessing the goal here is mostly to have a dumb timeout? If that's the case I wonder if a `for _ in range(...)` loop might be clearer (though admittedly slightly harder to configure)?